flink-ml-parent/flink-ml-lib/src/test/java/org/apache/flink/ml/common/statistics/basicstatistic/MultivariateGaussianTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.statistics.basicstatistic;
+
+import org.apache.flink.ml.common.linalg.DenseMatrix;
+import org.apache.flink.ml.common.linalg.DenseVector;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test cases for MultivariateGaussian.
+ */
+public class MultivariateGaussianTest {
+	private static final double TOL = 1.0e-5;
```

Comment:
Is there a reason why we use a fixed `TOL` while during the actual execution we use the limited bi-section to determine the actual TOL for the system?

Reply:
The limited bi-section is used to compute the threshold in determing whether the singular value of the covariance matrix is larger than zero. It is a way to improve numerical stability when the covariance matrix is singular or nearly singular. The "TOL" in the unit test cases is the tolerance between the computed pdf and theoretical pdf. We could only  achieve the precision at around 1*e-5, because det(sigma) is computed with LAPACK's "dsyev", which uses iterative algorithm to compute eigen values, thus have inherent errors.